src/pybind/mgr/prometheus/module.py

```diff
@@ -263,7 +263,10 @@ def get_pg_status(self):
                          key.split('+')]
         for state, value in reported_pg_s:
             path = 'pg_{}'.format(state)
-            self.metrics[path].set(value)
+            if path in self.metrics:
+                self.metrics[path].set(value)
```

Comment:
Might be a little bit more pythonic+efficient to do this by doing a .get() and catching KeyError instead of doing the `in` test first

Reply:
Ah yes of course...thx

Reply:
oh my bad... .get() never throws a KeyError. So access via .[path] is needed to get the exception.